### DISCUSSION TITLE
How to translate text into English image?

### DISCUSSION THREAD

**Discussion by mrcego** [tony19/vite-docs-template#5] [Q&A] [answered]
According commit [#d92d7f6](https://github.com/tony19/vite-docs-template/commit/d92d7f6c9e1b3b9055ec0f872fd545c9fa69a10b), we should translate English text  into images in docs. How to do that? Any guide about this? I'm interested.

**Comment by tony19** [comment]
I'm wondering myself. 🙂 I'll have to update the steps with instructions once I figure it out.

**Reply by tony19** [reply]
It looks like Evan created the PNGs a while back (`{bundler,esm,graph}.png`). I think we should convert them to SVGs to enable translations.

**Reply by mrcego** [reply]
What about [this kind of images](https://github.com/vitejs/vite/blob/main/docs/images/v3-new-open-issues-and-PRs.png)? I was testing some tools to parse SVG path from images and standards isn't the same at all. Trying to edit SVG Path file later was ponderous and need time. IMHO, there are two scenarios here:

1. Convince Evan to create shareable SVG files with a tool of his preference and test several tools to edit that files, or
2. Create a tool satisfying both issues (shareable and editable SVG files with global standard). 

While we are discussing this, I will continue to search and test the correct tools to achieve this with old images.

**Reply by tony19** [reply]
I think even those images (which are just graphs with text) can be recreated in SVG. Once we convert those PNGs to SVG, we could easily maintain the translations in the SVGs going forward. We don't need to create a tool to auto convert the PNGs (that would be overkill IMO).

**Reply by mrcego** [reply]
Yeah, I suggested second option just in case with.

I have tested Adobe Express online tool and convert images with an OK fidelity quality. Now, I'm searching a graphic editor tool for generated SVG.

**Comment by tony19** [comment, marked as answer]
I re-drew the PNGs into SVGs with [Figma](https://www.figma.com). Translations can be applied to the SVGs themselves by editing them directly. If you need to expand boxes for your translation to fit the text, you could also use Figma or any other SVG editor to adjust the diagram.

related: https://github.com/vitejs/vite/pull/9208

**Reply by mrcego** [reply]
I was playing a little, trying to understand how to vectorize PNG's. Some advices to get SVG files with  at least a similar quality from source?

**Reply by tony19** [reply]
I'm not aware of a tool that creates perfect SVGs from PNGs. There's a Figma plugin (Vectorized) that traces the outlines of an image, but the output would have to be colored in.
